templates/grant.erb

```diff
@@ -2,5 +2,9 @@
 <% if @bool_mysql_create_db -%>
 CREATE DATABASE IF NOT EXISTS <%= @real_db %><%= @real_db_create_options %>;
 <% end -%>
-GRANT <%= @mysql_privileges %> ON <%= @real_db %>.* TO '<%= @mysql_user %>'@'<%= @mysql_host %>' IDENTIFIED BY '<%= @mysql_password %>'<% if @require_ssl  -%> require ssl<% end -%>;
+GRANT <%= @mysql_privileges 
+  %> ON <%= @real_db 
+  %>.* TO '<%= @mysql_user %>'@'<%= @mysql_host 
+  %>' IDENTIFIED BY PASSWORD '<%= "*" + Digest::SHA1.hexdigest(Digest::SHA1.digest(@mysql_password)).upcase
```

Comment:
It is not compatible with mysql pre-4.1 
The ruby hash will calculate the password string which responds to password() function in mysql. Password() is deprecated in mysql 5.7 so I am not sure it is future compatible.
Hexdigest of a digest is because mysql is calculating sha1 twice for password.

https://jsosic.wordpress.com/tag/generate-mysql-password/